.gitignore

```diff
@@ -1,46 +1,395 @@
-/.lock-waf*
-/.waf*
-/Tools/autotest/aircraft/Rascal/reset.xml
-/Tools/autotest/ch7_mission.txt
-/Tools/autotest/jsb_sim/fgout.xml
-/Tools/autotest/jsb_sim/rascal_test.xml
-/Tools/autotest/jsbsim_fgout_0.xml
-/Tools/autotest/jsbsim_start_0.xml
+## Common
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+*.elf
+
+# Compiled Dynamic libraries
+*.so
+*.so.*
+*.dylib
+*.dll
+
+# Compiled Static libraries
+*.lai
+*.la
+*.a
+*.lib
+
+# Executables
+*.exe
+*.out
+*.app
+*.i*86
+*.x86_64
+*.hex
+
+#ignore merge backup file
+*.orig
+
+## Windows ##
+# Windows image file caches
+Thumbs.db
+ehthumbs.db
+
+# Folder config file
+Desktop.ini
+
+# Recycle Bin used on file shares
+$RECYCLE.BIN/
+
+# Windows Installer files
+*.cab
+*.msi
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
+
+#Ignore files build by Visual Studio
```

Comment:
because of that : http://ardupilot.org/dev/docs/building-ardupilot-apm-with-visual-studio-visual-micro.html ... but is only for avr apm ...